packages/store-ui/src/molecules/Modal/useTrapFocus.ts

```diff
@@ -0,0 +1,139 @@
+import { useEffect, useRef } from 'react'
+import type { RefObject } from 'react'
+import type { FocusableElement } from 'tabbable'
+import { tabbable } from 'tabbable'
+
+interface TrapFocusParams {
+  trapFocusRef: RefObject<HTMLElement>
+}
+
+/*
+ * Element that will maintain the focus inside trapFocusRef, focus the first element,
+ * and focus back on the element that was in focus when useTrapFocus was triggered.
+ *
+ * Inspired by Reakit useTrapFocus https://github.com/reakit/reakit/blob/a211d94da9f3b683182568a56479b91afb1b85ae/packages/reakit/src/Dialog/__utils/useFocusTrap.ts
+ */
+const useTrapFocus = ({ trapFocusRef }: TrapFocusParams) => {
+  // #L31
+  const tabbableNodesRef = useRef<FocusableElement[]>()
+
+  const beforeElementRef = useRef<HTMLDivElement | null>(null)
+  const afterElementRef = useRef<HTMLDivElement | null>(null)
+  const nodeToRestoreRef = useRef<HTMLElement | null>(
+    document.hasFocus() ? (document.activeElement as HTMLElement) : null
+  )
+
+  // Add before and after elements, and set refs.
+  useEffect(() => {
+    if (!trapFocusRef.current) {
+      return
+    }
+
+    if (!beforeElementRef.current) {
+      const beforeElement = document.createElement('div')
+
+      beforeElement.tabIndex = 0
+      beforeElement.setAttribute('aria-hidden', 'true')
+      beforeElement.setAttribute('data-testid', 'beforeElement')
+      beforeElementRef.current = beforeElement
+    }
+
+    if (!afterElementRef.current) {
+      afterElementRef.current = beforeElementRef.current.cloneNode() as HTMLDivElement
+      afterElementRef.current.setAttribute('data-testid', 'afterElement')
+    }
+
+    trapFocusRef.current.insertAdjacentElement(
+      'beforebegin',
+      beforeElementRef.current
+    )
+
+    trapFocusRef.current.insertAdjacentElement(
+      'afterend',
+      afterElementRef.current
+    )
```

Comment:
I didn't get why it won't render on SSR? Take a look here https://github.com/vtex-sites/base.store/pull/46 .
As far as I know, this code will only run on the client since the effect runs after the first render.

I did this about creating the elements inside the hook because you suggest that this creation be transparent for the user.

Reply:
Removed, bring back the approach to receive it via props.